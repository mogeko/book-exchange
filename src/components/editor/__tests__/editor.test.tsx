import "@testing-library/jest-dom";
import { render, screen, fireEvent } from "@testing-library/react";
import Editor from "@/components/editor/editor";

describe("Editor", () => {
  it("render an Editor", () => {
    const { container } = render(
      <Editor onSubmit={(data) => expect(data.content).toBeDefined()} />
    );

    expect(screen.getByRole("textbox")).toBeInTheDocument();
    expect(screen.getByRole("textbox").querySelector("p")).toHaveAttribute(
      "data-placeholder"
    );
    expect(screen.getByRole("button", { name: /submit/i })).toBeInTheDocument();
    expect(screen.getByText(/0 characters/i)).toBeInTheDocument();

    fireEvent.click(screen.getByRole("button", { name: /submit/i }));

    expect(container).toMatchSnapshot();
  });
});
