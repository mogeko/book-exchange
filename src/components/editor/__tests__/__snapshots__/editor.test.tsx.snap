// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor render an Editor 1`] = `
<div>
  <div
    class="relative max-w-xl"
  >
    <div
      class="flex h-40 w-full"
    >
      <div
        class="ProseMirror textarea textarea-bordered textarea-accent h-full w-full overflow-auto prose"
        contenteditable="true"
        role="textbox"
        tabindex="0"
        translate="no"
      >
        <p
          class="is-empty is-editor-empty"
          data-placeholder="Write something here..."
        >
          <br
            class="ProseMirror-trailingBreak"
          />
        </p>
      </div>
    </div>
    <div
      class="label items-start"
    >
      <span
        class="label-text-alt"
      >
        0
         characters
      </span>
      <button
        class="btn btn-sm btn-primary"
        disabled=""
      >
        Submit
      </button>
    </div>
  </div>
</div>
`;
